diff --git a/lib/javascript/typescript-config.d.ts b/lib/javascript/typescript-config.d.ts
index 292b0074351a61dd3b1786bc1701f15b7d6d2543..c595f99fb655943d539442741717b07f5f9f9a33 100644
--- a/lib/javascript/typescript-config.d.ts
+++ b/lib/javascript/typescript-config.d.ts
@@ -556,7 +556,7 @@ export interface TypeScriptCompilerOptions {
      *
      * @see https://www.typescriptlang.org/tsconfig#allowUnusedLabels
      */
-    readonly allowUnusedLabels?: false;
+    readonly allowUnusedLabels?: boolean;
     /**
      * Allow Unreachable Code
      *
@@ -570,7 +570,7 @@ export interface TypeScriptCompilerOptions {
      *
      * @see https://www.typescriptlang.org/tsconfig#allowUnreachableCode
      */
-    readonly allowUnreachableCode?: false;
+    readonly allowUnreachableCode?: boolean;
     /**
      * Check JS
      *
@@ -580,7 +580,7 @@ export interface TypeScriptCompilerOptions {
      *
      * @see https://www.typescriptlang.org/tsconfig#checkJs
      */
-    readonly checkJs?: true;
+    readonly checkJs?: boolean;
 }
 /**
  * Container for `TypescriptConfig` `tsconfig.json` base configuration(s).
