diff --git a/lib/javascript/typescript-config.d.ts b/lib/javascript/typescript-config.d.ts
index 2331b8a6f7977d9b29a345b68f88131d141d4a88..122aecf5bd9a65076e4813f9117fc6d81f91ab15 100644
--- a/lib/javascript/typescript-config.d.ts
+++ b/lib/javascript/typescript-config.d.ts
@@ -282,7 +282,7 @@ export interface TypeScriptCompilerOptions {
      *
      * @default false
      */
-    readonly emitDeclarationOnly?: true;
+    readonly emitDeclarationOnly?: boolean;
     /**
      * Do not emit compiler output files like JavaScript source code, source-maps or
      * declarations if any errors were reported.