diff --git a/dist/index.mjs b/dist/index.mjs
index ce4e57edc8ab77c768647c385f52c9e9751b05c2..b318027b6e2f0d0203d089cfa9cd96e21af64056 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -642,6 +642,24 @@ async function rollupBuild(ctx) {
           )};` : ""
         ].join("\n")
       );
+      await writeFile(
+        output + ".d.mts",
+        [
+          `export * from ${JSON.stringify(resolvedEntryWithoutExt)};`,
+          hasDefaultExport ? `export { default } from ${JSON.stringify(
+            resolvedEntryWithoutExt
+          )};` : ""
+        ].join("\n")
+      );
+      await writeFile(
+        output + ".d.cts",
+        [
+          `export * from ${JSON.stringify(resolvedEntryWithoutExt)};`,
+          hasDefaultExport ? `export { default } from ${JSON.stringify(
+            resolvedEntryWithoutExt
+          )};` : ""
+        ].join("\n")
+      );
       if (shebang) {
         await makeExecutable(output + ".cjs");
         await makeExecutable(output + ".mjs");